### PR TITLE
fix: update system_content with updated strings and always return str

### PR DESCRIPTION
**Pull request by onerandomusername**
## Summary

fixes an issue where system_content could potentially be None even when documented as returning a string in all cases.
Additionally updates the strings to the latest strings from the client. There's a few cases which have questions which are current documented as comments, and I was not sure about app commands and context menu commands. 

This pr is waiting on having the above questions answered before documentation and changelogs are added.

## Checklist

- [ ] If code changes were made, then they have been tested
    - [ ] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint`
    - [x] I have type-checked the code by running `task pyright`
- [x] This PR fixes an issue

### CODE REVIEW

**Review by shiftinv — CHANGES_REQUESTED** [review]

File: disnake/message.py [hunk 1]
```diff
@@ -1261,16 +1261,18 @@ def is_system(self) -> bool:
...
+    def system_content(self) -> str:
         """
         :class:`str`: A property that returns the content that is rendered
         regardless of the :attr:`Message.type`.
 
         In the case of :attr:`MessageType.default` and :attr:`MessageType.reply`\\,
         this just returns the regular :attr:`Message.content`. Otherwise this
         returns an English message denoting the contents of the system message.
+
+        If the message type is unrecognised this method will return an empty string.
```
[the remark this answers is not in the document]
Comment (onerandomusername) [reply disnake/message.py:1273]: That's fair.

File: disnake/message.py [hunk 2]
```diff
@@ -1339,11 +1345,11 @@ def system_content(self):
                 return f"{self.author.name} just boosted the server **{self.content}** times! {self.guild} has achieved **Level 3!**"
 
         if self.type is MessageType.channel_follow_add:
-            return f"{self.author.name} has added {self.content} to this channel"
+            return f"{self.author.name} has added {self.content} to this channel. Its most important updates will show up here."
 
         if self.type is MessageType.guild_stream:
```
[the remark this answers is not in the document]
Comment (onerandomusername) [reply disnake/message.py:1354]: This message is still in the client.

File: disnake/message.py [hunk 3]
```diff
@@ -1371,8 +1379,18 @@ def system_content(self):
...
 
         if self.type is MessageType.guild_invite_reminder:
+            # todo: determine if this should be the owner content or the user content
             return "Wondering who to invite?\nStart by inviting anyone who can help you build the server!"
 
+        if self.type is MessageType.context_menu_command:
+            return self.content
+
+        if self.type is MessageType.auto_moderation_action:
+            return self.content
```
[the remark this answers is not in the document]
Comment (onerandomusername) [reply disnake/message.py:1393]: Should we return None?

[the remark this answers is not in the document]
Comment (onerandomusername) [reply disnake/message.py:1393]: https://github.com/DisnakeDev/disnake/pull/766/commits/f7858906ce38b15681b077f1f4a506cff7a99521

File: disnake/message.py [hunk 4]
```diff
@@ -1285,7 +1287,11 @@ def system_content(self):
             else:
                 return f"{self.author.name} removed {self.mentions[0].name} from the thread."
 
+        # MessageType.call cannot be read by bots.
+
         if self.type is MessageType.channel_name_change:
+            if self.channel.type is ChannelType.public_thread and self.channel.parent.type is ChannelType.forum:  # type: ignore
```
[the remark this answers is not in the document]
Comment (onerandomusername) [reply disnake/message.py]: https://github.com/DisnakeDev/disnake/pull/766/commits/0685583309af5f9bda17ca1e3f11de3aceb9cd45

File: disnake/message.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (onerandomusername) [reply disnake/message.py]: https://github.com/DisnakeDev/disnake/pull/766/commits/f7858906ce38b15681b077f1f4a506cff7a99521

File: disnake/message.py [hunk 5]
```diff
@@ -1371,8 +1383,18 @@ def system_content(self):
             return self.reference.resolved.content  # type: ignore
 
         if self.type is MessageType.guild_invite_reminder:
+            # todo: determine if this should be the owner content or the user content
```
Comment (shiftinv) [inline disnake/message.py:1386]: This should be fine for now, anything else would have to rely on checking guild ownership and permissions.

Comment (onerandomusername) [reply disnake/message.py:1386]: I still would like to leave this todo in here.

**Review by shiftinv — COMMENTED** [review]
Would you mind adding a changelog entry for this?

**Comment by onerandomusername** [comment]
Added.

File: changelog/766.breaking.rst [hunk 6]
```diff
@@ -0,0 +1 @@
+Properly document that :attr:`Message.system_content` may return None. While this is documented as a breaking change, this function always could return None if the message type was not recognised.
```
Comment (shiftinv) [inline changelog/766.breaking.rst:1]:
```suggestion
Properly document that :attr:`Message.system_content` may return ``None``. While this is documented as a breaking change, this function always could return ``None`` if the message type was not recognised.
```

**Review by shiftinv — APPROVED** [review]

**Pull request merged**